rest/route/github.go

```diff
@@ -603,90 +606,83 @@ func (gh *githubHookApi) AddIntentForPR(ctx context.Context, pr *github.PullRequ
-			err := gh.sc.AddCommentToPR(ctx, pr.Base.User.GetLogin(), pr.Base.Repo.GetName(), pr.GetNumber(), overridingPatchComment)
-			if err != nil {
-				errs = append(errs, err)
-			}
+	// If we do want to override any existing patches, we override them and then create a new patch.
+	err = gh.overrideOtherPRs(ctx, pr, ghi, conflictingPatches)
+	if err != nil {
+		return errors.Wrap(err, "overriding other PRs")
+	}
 
-			if len(errs) > 0 {
-				errMsg := ""
-				for _, err := range errs {
-					errMsg += err.Error() + "\n"
-				}
-				grip.Info(message.Fields{
-					"message": "error informing user that aborting other patches failed",
-					"owner":   pr.Base.User.GetLogin(),
-					"repo":    pr.Base.Repo.GetName(),
-					"ref":     pr.Head.GetRef(),
-					"pr_num":  pr.GetNumber(),
-					"err":     errMsg,
-				})
-			}
-		}
+	return errors.Wrap(data.AddPatchIntent(ghi, gh.queue), "saving GitHub patch intent")
+}
+
+// overrideOtherPRs aborts all the patches in the list and comments on the PR's that they were aborted.
+// As well as comments on the given PR that it is overriding the other patches.
+func (gh *githubHookApi) overrideOtherPRs(ctx context.Context, pr *github.PullRequest, intent patch.Intent, patches []patch.Patch) error {
+	// If we want to override existing patches, we need to abort them and create a new patch.
+	// While informing all the PR's that their patch was aborted in favor of the new one.
+	grip.Info(message.Fields{
+		"message":         "aborting existing CI on same SHA patches",
+		"owner":           pr.Base.User.GetLogin(),
+		"repo":            pr.Base.Repo.GetName(),
+		"ref":             pr.Head.GetRef(),
+		"pr_num":          pr.GetNumber(),
+		"conflicting_prs": gh.comments.getLinksForPRPatches(patches),
+	})
+
+	// Comment on the PR's that were aborted and then abort them.
+	// If these error, we should should still continue and create a patch as the others were aborted.
+	catcher := grip.NewBasicCatcher()
+	for _, p := range patches {
+		catcher.Add(gh.sc.AddCommentToPR(ctx, p.GithubPatchData.BaseOwner, p.GithubPatchData.BaseRepo, p.GithubPatchData.PRNumber, gh.comments.overriddenPR(pr)))
 	}
 
-	if err := data.AddPatchIntent(ghi, gh.queue); err != nil {
-		return errors.Wrap(err, "saving GitHub patch intent")
+	catcher.Add(gh.sc.AddCommentToPR(ctx, pr.Base.User.GetLogin(), pr.Base.Repo.GetName(), pr.GetNumber(), gh.comments.overriddingPR(patches)))
+	grip.Error(message.WrapError(catcher.Resolve(), message.Fields{
+		"message": "error informing user(s) that aborting other patches failed",
+		"owner":   pr.Base.User.GetLogin(),
+		"repo":    pr.Base.Repo.GetName(),
+		"ref":     pr.Head.GetRef(),
+		"pr_num":  pr.GetNumber(),
+	}))
+
+	// Abort patches, and if any fail, comment on the PR to inform the user and no-op.
+	for _, p := range patches {
```

Comment:
May be a little neater to abort patches in the same loop as above, just after adding the comment. Would also make sense to continue on error in case any single patch fails to abort, since we still want to try aborting later PR patches in the slice.

Reply:
I combined them and made a comment for the function to explain the design choice. I also put two grip logging, one for comments and one for patch cancels, with different levels since erroring on comments is not as important as erroring on patches.

I could see a case where it might be good to error on comments as well for information, if  you think so I could change it so they are treated the same.

Reply:
Eh, splitting the error logs seems fine. LGTM!